### PR TITLE
feat(cypress): add cypress to the compliance

### PR DESCRIPTION
**Pull request by Fewwy**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist
Adding cypress testing library to the project. So anyone could write cypress tests when they need to
## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: src/SmartComponents/Reports/Reports.cy.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+  it('expect to render emptystate', () => {
+    Cypress.config('interceptions', {
+      data: {
+        profiles: { edges: [] },
+      },
+      error: false,
+      loading: false,
+    });
+    cy.get('table').should('have.length', 1);
```
Comment (mkholjuraev) [inline src/SmartComponents/Reports/Reports.cy.js:89]: I think we can improve this test by checking if there is an empty state component as well inside the table. Now the test only verifies if there is a table.

File: package.json [hunk 2]
```diff
@@ -8,6 +8,8 @@
   "dependencies": {
     "@apollo/client": "3.4.16",
     "@babel/runtime": "^7.19.4",
+    "@cypress/webpack-dev-server": "^3.1.2",
+    "@patternfly/patternfly": "^4.222.4",
```
Comment (mkholjuraev) [inline package.json:12]: Should not this be installed as a dev-dependency?

Comment (Fewwy) [reply package.json:12]: Good catch, thanks will do that

**Review by gkarat — COMMENTED** [review]
Can we also integrate cypress to code coverage within this PR already? :-)

File: package.json [hunk 3]
```diff
@@ -82,6 +82,21 @@
...
   },
+  "nyc": {
+    "report-dir": "cypress-coverage",
+    "include": [
+      "src/Mutations",
+      "src/PresentationalComponents",
+      "src/Modules",
+      "src/SmartComponents",
+      "src/__factories__",
+      "src/Utilities"
```
Comment (gkarat) [inline package.json:93]: Is there a reason to omit other files under `src`? Can we use just:
```suggestion
      "src/**/*"
```

File: package.json [hunk 4]
```diff
@@ -82,6 +82,21 @@
...
+    ],
+    "exclude": [
+      "src/**/*.spec.ct.js",
```
Comment (gkarat) [inline package.json:96]: This will exclude all cypress test files from coverage 🤔 we should remove this entry
```suggestion
```

**Review by gkarat — CHANGES_REQUESTED** [review]

**Pull request closed without merging**